persistence/memorypersistence/aggregate.go

```diff
@@ -46,23 +64,61 @@ func (v *validator) VisitSaveAggregateMetaData(
 	}
 }
 
+// VisitSaveAggregateSnapshot returns an error if a "SaveAggregateSnapshot"
+// operation can not be applied to the database.
+func (v *validator) VisitSaveAggregateSnapshot(
+	_ context.Context,
+	op persistence.SaveAggregateSnapshot,
+) error {
+	ss := op.Snapshot
+
+	if ss.HandlerKey != "" && ss.InstanceID != "" {
+		return nil
+	}
+
+	return persistence.NotFoundError{
+		Cause: op,
+	}
+}
+
+// VisitRemoveAggregateSnapshot returns an error if a "RemoveAggregateSnapshot"
+// operation can not be applied to the database.
+func (v *validator) VisitRemoveAggregateSnapshot(
+	_ context.Context,
+	op persistence.RemoveAggregateSnapshot,
+) error {
+	inst := op.Snapshot
+	key := instanceKey{inst.HandlerKey, inst.InstanceID}
+
+	if x, ok := v.db.aggregate.snapshot[key]; ok {
+		if inst.LastEventID == x.LastEventID {
+			return nil
+		}
+	}
+
+	return persistence.NotFoundError{
+		Cause: op,
+	}
+}
```

Comment:
Much the same as for `SaveAggregateSnapshot`, there's no semantic reason to reject a remove operation, so no validation is required.

I note you are checking the last event ID, which I guess serves as a kind of confirmation that you are deleting "this snapshot", but the `RemoveAggregateSnapshot` operation can be thought of as "make sure there is no snapshot of this instance" at all, since we only ever need a single snapshot.

```suggestion
	return nil
}
```